src/test/java/net/datafaker/base/PhoneNumberTest.java

```diff
@@ -121,11 +125,89 @@ void testPhone_CA() {
+            }
+            assertThat(errorCount).isLessThan(250);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("generateLanguageAndRegionOfLocales")
+    void testAllPhoneNumberInternational(String localeLanguage, Map<String, String> regions) throws NumberParseException {
+        final PhoneNumberUtil util = PhoneNumberUtil.getInstance();
+        for (Map.Entry<String, String> region : regions.entrySet()) {
+            int errorCount = 0;
+            final BaseFaker faker = new BaseFaker(new Locale(localeLanguage, region.getKey()));
+            for (int i = 0; i < 10; i++) {
+                String phoneNumber = faker.phoneNumber().phoneNumberInternational();
+                Phonenumber.PhoneNumber proto = util.parse(phoneNumber, region.getValue());
+                if (!util.isValidNumberForRegion(proto, region.getValue())) {
+                    errorCount++;
+                }
+            }
+            assertThat(errorCount).isLessThan(250);
+        }
+    }
+    // Key("IND") in `map("IND", "IN")` is a region for Java's Locale class
+    // Value("IN") in `map("IND", "IN")` is a region for PhoneNumberUtil's `parse` method.
+    private static Stream<Arguments> generateLanguageAndRegionOfLocales() {
+        return Stream.of(
+            Arguments.of("en", map("US", "US")),
+            Arguments.of("en", map("GB", "GB")),
+            Arguments.of("en", map("AU", "AU")),
+            Arguments.of("en", map("CA", "CA")),
+            Arguments.of("en", map("MS", "MS")),
+            Arguments.of("en", map("NG", "NG")),
+            Arguments.of("en", map("NZ", "NZ")),
+            Arguments.of("bg", map("BG", "BG")),
+            Arguments.of("by", map("BY", "BY")),
+            Arguments.of("ca", map("CA", "CA")),
+            Arguments.of("cs", map("CZ", "CZ")),
+            Arguments.of("de", map("DE", "DE")),
+            Arguments.of("de", map("AT", "AT")),
+            Arguments.of("de", map("CH", "CH")),
+            Arguments.of("en", map("IND", "IN")),
+            Arguments.of("en", map("NEP", "NP")),
+            Arguments.of("en", map("PAK", "PK")),
+            Arguments.of("hu", map("HU", "HU")),
+            Arguments.of("fi", map("FI", "FI")),
+            Arguments.of("ko", map("KR", "KR")),
+            Arguments.of("ja", map("JP", "JP")),
+            Arguments.of("lv", map("LV", "LV")),
+            Arguments.of("it", map("IT", "IT")),
+            Arguments.of("nl", map("NL", "NL")),
+            Arguments.of("pl", map("PL", "PL")),
+            Arguments.of("pt", map("PT", "PT")),
+            Arguments.of("zh", map("CN", "CN")),
+            Arguments.of("zh", map("TW", "TW")),
+            Arguments.of("uk", map("UA", "UA")),
+            Arguments.of("tr", map("TR", "TR"))
+        );
+    }
+
+    private static Map<String, String> map(String key, String value) {
+        Map<String, String> map = new HashMap<>();
+        map.put(key, value);
+        return map;
+    }
+
```

Comment:
```suggestion
    private static Stream<Arguments> generateLanguageAndRegionOfLocales() {
        return Stream.of(
            Arguments.of(new Locale("en", "US"), "US"),
            Arguments.of(new Locale("en", "GB"), "GB"),
            Arguments.of(new Locale("en", "AU"), "AU"),
            Arguments.of(new Locale("en", "CA"), "CA"),
            Arguments.of(new Locale("en", "MS"), "MS"),
            Arguments.of(new Locale("en", "NG"), "NG"),
            Arguments.of(new Locale("en", "NZ"), "NZ"),
            Arguments.of(new Locale("bg", "BG"), "BG"),
            Arguments.of(new Locale("by", "BY"), "BY"),
            Arguments.of(new Locale("ca", "CA"), "CA"),
            Arguments.of(new Locale("cs", "CZ"), "CZ"),
            Arguments.of(new Locale("de", "DE"), "DE"),
            Arguments.of(new Locale("de", "AT"), "AT"),
            Arguments.of(new Locale("de", "CH"), "CH"),
            Arguments.of(new Locale("en", "IND"), "IN"),
            Arguments.of(new Locale("en", "NEP"), "NP"),
            Arguments.of(new Locale("en", "PAK"), "PK"),
            Arguments.of(new Locale("hu", "HU"), "HU"),
            Arguments.of(new Locale("fi", "FI"), "FI"),
            Arguments.of(new Locale("ko", "KR"), "KR"),
            Arguments.of(new Locale("ja", "JP"), "JP"),
            Arguments.of(new Locale("lv", "LV"), "LV"),
            Arguments.of(new Locale("it", "IT"), "IT"),
            Arguments.of(new Locale("nl", "NL"), "NL"),
            Arguments.of(new Locale("pl", "PL"), "PL"),
            Arguments.of(new Locale("pt", "PT"), "PT"),
            Arguments.of(new Locale("zh", "CN"), "CN"),
            Arguments.of(new Locale("zh", "TW"), "TW"),
            Arguments.of(new Locale("uk", "UA"), "UA"),
            Arguments.of(new Locale("tr", "TR"), "TR")
        );
    }
```

Reply:
Already fixed, thank you for pointing at the problem